### ISSUE TITLE
Behavior Simulator 1.1

### ISSUE THREAD

**Issue by knoellle** [HULKs/hulk#170]
# New stuff

- [ ] Think about where to draw the line between simulator and scenario file/lua library.
  - Specifically regarding application of rules, game logic and physics. Most of this currently either isn't implemented or done by the simulator.
- [ ] Fix penalty handling
  - currently both the robots and the game controller state have a penalty state, both of which are used in different locations

# Leftovers from #91:

## Important

- [ ] Limit robot's field of view & view range
- [ ] Simulate node inputs that are not currently covered
  - [ ] obstacles (use actual node?)
  - [ ] fall state
  - [ ] penalty shot direction (use actual node?)
  - [ ] has ground contact
- [ ] HULK standard library in Lua
- [ ] Build & Run simulator in CI

## Nice to have

- [ ] Rerun simulation when
  - [ ] scenario file changes
  - [ ] changing parameters via communication
- [ ] Ball "physics"
- [ ] Add configurable noise to
  - [ ] ball position
- [ ] message delay/jitter/dropping
- [ ] Apply SPL rules
- [ ] Automatically generate behavior cycler code. Probably after #143

**Comment by knoellle** [comment]
Agreed, but those are probably candidates for a behavior simulator 2.0 and twix 2.0.

**Comment by h3ndrk** [comment]
As proposed by Patrick, we should have a look in the simulation leagues before implementing our own simulator.

**Comment by schmidma** [comment]
kann zu
